tests/test_flows.py

```diff
@@ -278,6 +279,18 @@ def run_tests(self, tests):
             self.check_error = 0
             try:
                 getattr(self, test_full)()
+            except AssertionError as ae:
+                # do not add empty message if test has already been marked as failed
+                # and AssertError does not contain a message
+                if self.check_error:
+                    if not str(ae):
+                        pass
+                else:
+                    self.fail("{}".format(ae))
```

Comment:
The True branch of the condition does nothing at all, so
```suggestion
                if not self.check_error:
                    self.fail("{}".format(ae))
```

However, I don't see why you would depend on check_error. If the assertion came after a completely unrelated failure, you still want to mark the failure. So I'd just do:

```suggestion
                if str(ae):
                    self.fail("{}".format(ae))
```

Reply:
I still wanted test to fail if self.fail() is not called before (inside test method) and there is no assertion failure message. I ended up printing a generic "Assertion failed" message in that case.